### ISSUE TITLE
Checkbox disappeared about a year ago

### ISSUE THREAD

**Issue by Rvervuurt** [stcr/subscribe-to-comments-reloaded#789]
Hi

A client is using your plugin, but the checkbox disappeared about a year ago. They hadn't really thought about it, but would like the possibility to start using it again, so I started researching.

When using the wayback-machine, I can see that the checbox is there halfway december 2022 (though I can't check if it's functional):
![image](https://github.com/stcr/subscribe-to-comments-reloaded/assets/1439765/d52fa8b9-b61b-4be6-b1a0-7f8075d10f19)

Another check in january 2023 (same article), and the checkbox is gone:
![image](https://github.com/stcr/subscribe-to-comments-reloaded/assets/1439765/2f681250-7c73-48df-a794-37ac4408bb2d)

I rolled back all my code to august 2022, and the checkbox is not there, so this makes me suspect that it's this plugin that changed something in january 2023? Can you assist?

**Comment by BobaWebDev** [comment]
Hi @Rvervuurt 

Can you send over the URL to the site? Seeing it might give me an idea as to what's going on.

**Comment by Rvervuurt** [comment]
Hi @BobaWebDev 

Sure, here's a link: https://dinero.dk/nyheder/2023-automatiseringen-tog-fart/

**Comment by BobaWebDev** [comment]
I'm not seeing anything odd comparing the source code of one of the posts on wayback when the plugin worked and the real/live post where it doesn't.

It's been a while, but do you know if any plugins have been installed during that period (first half of January 2023)? Do you maybe have a backup from sometime before 2023 which could be used to make some comparisons (difference in plugins and settings)?

**Comment by Rvervuurt** [comment]
Good idea, I'll try to disable plugins one by one locally, see if that makes it pop up again.

**Comment by BobaWebDev** [comment]
Might be easier to immediately disable all of them except StCR, if the checkbox shows up then it's a conflict with one of the plugins. Then enable them one by one until the checkbox disappears.

If it doesn't show up even with all plugins disabled then try switching to a different theme, if it shows up it's a theme conflict.

And let me know how that goes so we can figure out how to proceed.

**Comment by Rvervuurt** [comment]
Ok, I found out that a custom plugin was overruling how the comment form was built!

I'm going to see if I can activate the plugin again, or if I have to rebuild my form in any way. Thanks for the push in the right direction!

**Comment by Rvervuurt** [comment]
On a totally clean install, the fields also don't show up:
![image](https://github.com/stcr/subscribe-to-comments-reloaded/assets/1439765/487038a2-3f6f-462a-abb9-6ec3fa49299a)
WP version: 6.4.2
PHP version: 8.2

WP Debugger returns the following, which might be part of the issue?
```[17-Jan-2024 14:21:58 UTC] PHP Deprecated:  Creation of dynamic property stcr\wp_subscribe_reloaded::$salt is deprecated in /Users/robbertvervuurt/Herd/tester/wp-content/plugins/subscribe-to-comments-reloaded/wp_subscribe_reloaded.php on line 65```

**Comment by BobaWebDev** [comment]
It's just a warning, doesn't really affect anything.

But I just remembered something another user mentioned a looong time ago. Can you go to WP admin > StCR > Options, scroll down to the end, you'll see "StCR Unique Key".

Is there a value in that field?

In any case, whether there's value or not, click the "Generate new key" button. After that check if the issue is resolved.

**Comment by BobaWebDev** [comment]
Also, if that doesn't work out, on that same options page, near the top there's an option called "StCR Position", is that set to Yes or No? In any case, no matter what it is at the moment, try the other one.

**Comment by Rvervuurt** [comment]
Yes, perfect! Removing the key and then saving the page worked on my clean installation. After that, I generated a key and it was still working.

This also got it working locally on my project's website. Thanks!

Any idea why this happens?

And agreed that it was just a warning, just wanted to share in case it was helpful :D

**Comment by Rvervuurt** [comment]
Can confirm this also worked on the live-page.

**Comment by BobaWebDev** [comment]
And if you do a fresh install of WP (no content related to that live site, just a blank WP install) and install the plugin the checkbox does not show up?

**Comment by BobaWebDev** [comment]
I replicated the issue. Will figure out what's going on.

**Comment by BobaWebDev** [comment]
Found the problem. It's not actually related to the unique key field, it's a problem with the "Enable on post types" option.

There's some array flipping happening there so keys and values switch place, so when it checks if the current post type supports the subscriptions it returns false since it checks the values.

Thanks for pointing out the issue.

**Comment by Rvervuurt** [comment]
Amazing, glad I could help!
